### PR TITLE
DO NOT MERGE | EDU-1181: Add High-availability Namespace Coverage

### PR DESCRIPTION
**Pull request by fairlydurable**
DO NOT MERGE

Building out Multi-Region Namespace Coverage for pre-release

JIRA: https://temporalio.atlassian.net/browse/EDU-1181

### CODE REVIEW

File: docs/production-deployment/cloud/multi-region.mdx [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+
+Your regional DNS must be reachable through IP addresses local to your network.
+A subzone for `region.<tmprl_domain>` enables you to override the regional DNS to make sure traffic is routed internally.
+You can check the Namespace's active region using the Namespace record CNAME, which is public.
+
+To set up the DNS override, you override specific regions to target the relevant IP addresses (e.g. aws-us-west-1.region.tmprl.cloud to target 192.168.1.2).
+Using AWS, this can be done using a private hosted zone in Route53 for `region.<tmprl_domain>`.
+Link to that private zone to the VPCs you use for Workers.
+
+When your Workers connect to the Namespace, they first resolve the ` <ns>.<acct>.<tmprl_domain>` record, which then targets `<active>.region.<tmprl_domain>` using a CNAME, leading traffic to reach the internal IP you're using.
```
Comment (XaF) [inline docs/production-deployment/cloud/multi-region.mdx:438]:
```suggestion
When your Workers connect to the Namespace, they first resolve the ` <ns>.<acct>.<tmprl_domain>` record, which targets `<active>.region.<tmprl_domain>` using a CNAME. That second DNS resolution is overridden by your private zone, leading traffic to reach the internal IP you're using.
```

File: docs/production-deployment/cloud/multi-region.mdx
Comment (XaF) [inline docs/production-deployment/cloud/multi-region.mdx:1]: I don't know where we'd want to add this but, here's a table of regions, private link endpoint, and record to override:

```markdown
| Region | PrivateLink VPC Endpoint | DNS Record Override |
|---|---|---|
| `ap-northeast-1` | `com.amazonaws.vpce.ap-northeast-1.vpce-svc-08f34c33f9fb8a48a` | `aws-ap-northeast-1.region.tmprl.cloud` |
| `ap-northeast-2` | `com.amazonaws.vpce.ap-northeast-2.vpce-svc-08c4d5445a5aad308` | `aws-ap-northeast-2.region.tmprl.cloud` |
| `ap-south-1` | `com.amazonaws.vpce.ap-south-1.vpce-svc-0ad4f8ed56db15662` | `aws-ap-south-1.region.tmprl.cloud` |
| `ap-southeast-1` | `com.amazonaws.vpce.ap-southeast-1.vpce-svc-05c24096fa89b0ccd` | `aws-ap-southeast-1.region.tmprl.cloud` |
| `ap-southeast-2` | `com.amazonaws.vpce.ap-southeast-2.vpce-svc-0634f9628e3c15b08` | `aws-ap-southeast-2.region.tmprl.cloud` |
| `ca-central-1` | `com.amazonaws.vpce.ca-central-1.vpce-svc-080a781925d0b1d9d` | `aws-ca-central-1.region.tmprl.cloud` |
| `eu-central-1` | `com.amazonaws.vpce.eu-central-1.vpce-svc-073a419b36663a0f3` | `aws-eu-central-1.region.tmprl.cloud` |
| `eu-west-1` | `com.amazonaws.vpce.eu-west-1.vpce-svc-04388e89f3479b739` | `aws-eu-west-1.region.tmprl.cloud` |
| `eu-west-2` | `com.amazonaws.vpce.eu-west-2.vpce-svc-0ac7f9f07e7fb5695` | `aws-eu-west-2.region.tmprl.cloud` |
| `sa-east-1` | `com.amazonaws.vpce.sa-east-1.vpce-svc-0ca67a102f3ce525a` | `aws-sa-east-1.region.tmprl.cloud` |
| `us-east-1` | `com.amazonaws.vpce.us-east-1.vpce-svc-0822256b6575ea37f` | `aws-us-east-1.region.tmprl.cloud` |
| `us-east-1` | `com.amazonaws.vpce.us-east-1.vpce-svc-0b06bbbd81fd41222` | `aws-us-east-1.region.tmprl.cloud` |
| `us-east-2` | `com.amazonaws.vpce.us-east-2.vpce-svc-01b8dccfc6660d9d4` | `aws-us-east-2.region.tmprl.cloud` |
| `us-west-2` | `com.amazonaws.vpce.us-west-2.vpce-svc-0473f53028c65d8fb` | `aws-us-west-2.region.tmprl.cloud` |
| `us-west-2` | `com.amazonaws.vpce.us-west-2.vpce-svc-0f44b3d7302816b94` | `aws-us-west-2.region.tmprl.cloud` |
```

File: docs/production-deployment/cloud/multi-region.mdx [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+- When a Namespace fails over to a standby region, Workers will be communicating cross-region.
+
+- In case of a complete regional outage, Workers in the original region may go down as well as the original Namespace. 
+  To keep Workflows moving during this level of outage, deploy a redundant set of Workers to your standby region.
+
+:::
+
+### Routing {#routing}
+
+When you enable multi-region for a Namespace, we replace a Namespace's DNS record `<ns>.<acct>.<tmprl_domain>` with a CNAME to a regional DNS record in the format `<region>.region.<tmprl_domain>`.
```
[the remark this answers is not in the document]
Comment (fairlydurable) [reply docs/production-deployment/cloud/multi-region.mdx:398]: Adopted as suggested but I broke the final clause into its own sentence to limit sentence length. Thank you!

File: docs/production-deployment/cloud/multi-region.mdx [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+
+This regional DNS record lives in a subzone: 
+
+```
+region.<tmprl_domain>
+```
```
[the remark this answers is not in the document]
Comment (fairlydurable) [reply docs/production-deployment/cloud/multi-region.mdx:404]: Deleting these lines then and seeing if there's a better place to move them.

File: docs/production-deployment/cloud/multi-region.mdx [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+
+Clients resolving the Namespace’s DNS record are directed to the regional DNS record.
```
[the remark this answers is not in the document]
Comment (fairlydurable) [reply docs/production-deployment/cloud/multi-region.mdx:406]: Uppercased N for namespace and adopted. Thanks

File: docs/production-deployment/cloud/multi-region.mdx [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+- At failover, routing is completely transparent for multi-region customers _unless_ you use PrivateLink.
+  This section describes how to configure routing for multi-region Namespaces for PrivateLink customers only.
+
+:::
+
+PrivateLink customers may need to change certain configurations for multi-region Namespace use.
+Routing configuration depends on networking setup and use of PrivateLink.
+You may need to:
+
+* set an override DNS; and
```
[the remark this answers is not in the document]
Comment (fairlydurable) [reply docs/production-deployment/cloud/multi-region.mdx:425]: Adopted. Thanks

File: docs/production-deployment/cloud/multi-region.mdx [hunk 6]
```diff
@@ -0,0 +1,447 @@
...
+You may need to:
+
+* set an override DNS; and 
+* ensure the network connectivity between the two regions.
+
+![Customer side solution example](/img/multi-region/private-link.png)
+
+Your regional DNS must be reachable through IP addresses local to your network.
+A subzone for `region.<tmprl_domain>` enables you to override the regional DNS to make sure traffic is routed internally.
+You can check the Namespace's active region using the Namespace record CNAME, which is public.
```
[the remark this answers is not in the document]
Comment (fairlydurable) [reply docs/production-deployment/cloud/multi-region.mdx:432]: Updated

File: docs/production-deployment/cloud/multi-region.mdx (hunk 1 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/multi-region.mdx:438]: Updated!

File: docs/production-deployment/cloud/multi-region.mdx [hunk 7]
```diff
@@ -0,0 +1,447 @@
...
+
+To set up the DNS override, you override specific regions to target the relevant IP addresses (e.g. aws-us-west-1.region.tmprl.cloud to target 192.168.1.2).
+Using AWS, this can be done using a private hosted zone in Route53 for `region.<tmprl_domain>`.
+Link to that private zone to the VPCs you use for Workers.
```
[the remark this answers is not in the document]
Comment (fairlydurable) [reply docs/production-deployment/cloud/multi-region.mdx:436]: Adopted

File: docs/production-deployment/cloud/multi-region.mdx
Comment (fairlydurable) [reply docs/production-deployment/cloud/multi-region.mdx:1]: Great table. Added, along with a note that `sa-east-1` is not yet available for use with MRN.

**Comment by fairlydurable** [comment]
@XaF Thank you so much for the table and the great feedback. It really helped.

**Comment by fairlydurable** [comment]
Adopted into EDU-1180. Closing this draft ticket.

**Pull request closed without merging**